### PR TITLE
tests: net: convert to ztest

### PR DESCRIPTION
**Pull request by nniranjhana**
Replaced test points with ztest APIs wherever possible for files in **tests/net/**

JIRA: ZEP-2249

Signed-off-by: Niranjhana N <niranjhana.n@intel.com>

### CODE REVIEW

**Review by jukkar — CHANGES_REQUESTED** [review]
Just keep everything in main.c, I do not see much point separating the stuff to two different files.

**Comment by nniranjhana** [comment]
@jukkar okay, fixed the commit with the suggested changes.

File: tests/net/icmpv6/src/main.c [hunk 1]
```diff
@@ -72,51 +75,41 @@ static bool run_tests(void)
...
+	/**TESTPOINT: Check input*/
+	zassert_true(!(ret < 0 || handler_status != 0),
+			"Callback not called properly");
 
-	printk("ICMPv6 tests passed\n");
+	/**TESTPOINT: Check input*/
+	zassert_true(!(handler_called != 2), "Callbacks not called properly");
 
-	return true;
+	printk("ICMPv6 tests passed\n");
```
Comment (nashif) [inline tests/net/icmpv6/src/main.c]: remove this, the pass/fail comes from ztest.

File: tests/net/ip-addr/src/main.c [hunk 2]
```diff
@@ -223,137 +226,101 @@ static bool run_tests(void)
 
 	printk("IP address print tests passed\n");
```
Comment (nashif) [inline tests/net/ip-addr/src/main.c]: remove this

File: tests/net/ip-addr/src/main.c [hunk 3]
```diff
@@ -418,28 +385,18 @@ static bool run_tests(void)
...
+		"IPv4 wrong match failed");
 
-	if (!net_ipv4_addr_mask_cmp(iface, &match_addr)) {
-		printk("IPv4 match failed\n");
-		return false;
-	}
+	zassert_true(net_ipv4_addr_mask_cmp(iface, &match_addr),
+		"IPv4 match failed");
 
 	printk("IP address checks passed\n");
```
Comment (nashif) [inline tests/net/ip-addr/src/main.c]: remove this

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/net/icmpv6/src/main.c (hunk 1 above)
Comment (nniranjhana) [inline tests/net/icmpv6/src/main.c]: @nashif okay, done

File: tests/net/ip-addr/src/main.c (hunk 3 above)
Comment (nniranjhana) [inline tests/net/ip-addr/src/main.c]: done

File: tests/net/ip-addr/src/main.c (hunk 2 above)
Comment (nniranjhana) [inline tests/net/ip-addr/src/main.c]: done

**Review by jukkar — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by nniranjhana** [comment]
@jukkar request to review the last commit
@nashif the commits for this PR are done

**Review by jukkar — APPROVED** [review]
LGTM

**Review by nashif — APPROVED** [review]

**Pull request merged**